services/collaboration/pkg/internal/app/wopicontext.go

```diff
@@ -0,0 +1,86 @@
+package app
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+
+	appproviderv1beta1 "github.com/cs3org/go-cs3apis/cs3/app/provider/v1beta1"
+	userv1beta1 "github.com/cs3org/go-cs3apis/cs3/identity/user/v1beta1"
+	providerv1beta1 "github.com/cs3org/go-cs3apis/cs3/storage/provider/v1beta1"
+	ctxpkg "github.com/cs3org/reva/v2/pkg/ctx"
+	"github.com/golang-jwt/jwt/v4"
+	"google.golang.org/grpc/metadata"
+)
+
+type key int
+
+const (
+	wopiContextKey key = iota
+)
+
+type WopiContext struct {
+	AccessToken   string
+	FileReference providerv1beta1.Reference
+	User          *userv1beta1.User
+	ViewMode      appproviderv1beta1.ViewMode
+	EditAppUrl    string
+	ViewAppUrl    string
+}
+
+func WopiContextAuthMiddleware(app *DemoApp, next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		accessToken := r.URL.Query().Get("access_token")
+		if accessToken == "" {
+			fmt.Println("wopicontext", "accesstoken empty")
```

Comment:
I guess the logger components should be used instead of printing to console right away ....